kmip/tests/unit/pie/test_client.py

```diff
@@ -0,0 +1,598 @@
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_enter(self):
+        """
+        Test the result and effect of the enter method for the context
+        manager.
+        """
+        client = ProxyKmipClient()
+
+        self.assertFalse(client._is_open)
+        result = client.__enter__()
+        self.assertEqual(result, client)
+        self.assertTrue(client._is_open)
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_exit(self):
+        """
+        Test the result and effect of the exit method for the context
+        manager.
+        """
+        client = ProxyKmipClient()
+        client.__enter__()
+
+        self.assertTrue(client._is_open)
+        client.__exit__(None, None, None)
+        self.assertFalse(client._is_open)
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_context_manager(self):
+        """
+        Test that the KmipClient can be used by the with-statement as a
+        context manager.
+        """
+        with ProxyKmipClient() as client:
+            self.assertTrue(client._is_open)
+            client.proxy.open.assert_called_with()
+        self.assertFalse(client._is_open)
+        client.proxy.close.assert_called_with()
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_create(self):
+        """
+        Test that a symmetric key can be created with proper inputs and that
+        its UID is returned properly.
+        """
+        # Create the template to test the create call
+        algorithm = enums.CryptographicAlgorithm.AES
+        length = 256
+        algorithm_attribute = self.attribute_factory.create_attribute(
+            enums.AttributeType.CRYPTOGRAPHIC_ALGORITHM, algorithm)
+        length_attribute = self.attribute_factory.create_attribute(
+            enums.AttributeType.CRYPTOGRAPHIC_LENGTH, length)
+        mask_attribute = self.attribute_factory.create_attribute(
+            enums.AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+            [enums.CryptographicUsageMask.ENCRYPT,
+             enums.CryptographicUsageMask.DECRYPT])
+
+        attributes = [algorithm_attribute, length_attribute, mask_attribute]
+        template = obj.TemplateAttribute(attributes=attributes)
+
+        status = enums.ResultStatus.SUCCESS
+        result = results.CreateResult(
+            contents.ResultStatus(status),
+            uuid=attr.UniqueIdentifier(
+                'aaaaaaaa-1111-2222-3333-ffffffffffff'))
+
+        with ProxyKmipClient() as client:
+            client.proxy.create.return_value = result
+
+            uid = client.create(algorithm, length)
+            client.proxy.create.assert_called_with(
+                enums.ObjectType.SYMMETRIC_KEY, template)
+            self.assertIsInstance(uid, six.string_types)
```

Comment:
You could also check the return ID to make sure it equals the value from line 202.